src/tike/cluster.py

```diff
@@ -158,6 +171,80 @@ def by_scan_stripes(
     ]
 
 
+def _sequential(
+    functions: typing.List,
+    population: npt.ArrayLike,
+    nums_cluster: typing.List[int],
+) -> typing.List[npt.NDArray]:
+    """Return indices dividing the population by applying functions sequentially.
+
+    The returned clusters are divided along the provided dimension into
+    clusters of approximate equal numbers of elements.
+
+    Parameters
+    ----------
+    functions : [f(population, num_cluster), f(population, num_cluster), ...]
+        A list of clustering functions to apply sequentially.
+    population : (M, N) array_like
+        The M samples of an N dimensional population that needs to be
+        clustered.
+    num_cluster : [int, int, ...]
+        The number of clusters in which to divide M samples at each step.
+
+    Returns
+    -------
+    indicies : (num_cluster,) list of array of integer
+        The indicies of population that belong to each cluster.
+    """
+    indices = functions[0](population, nums_cluster[0])
+
+    if len(functions) == 1:
+        return indices
+
+    for set in indices:
+        sub_indices = _sequential(
+            functions[1:],
+            population=population[set],
+            nums_cluster=nums_cluster[1:],
+        )
+        indices[sub_indices]
+
+
+def stripes_equal_count(
+    population: npt.ArrayLike,
+    num_cluster: int,
+    dim: int = 0,
+) -> typing.List[npt.NDArray]:
+    """Return indices dividing the population into stripes of equal count.
+
+    The returned clusters are divided along the provided dimension into
+    clusters of approximate equal numbers of elements.
+
+    Parameters
+    ----------
+    population : (M, N) array_like
+        The M samples of an N dimensional population that needs to be
+        clustered.
+    num_cluster : int (0..M]
+        The number of clusters in which to divide M samples.
+    dim : int
+        The dimension (of N) along which the population is divided.
+
+    Returns
+    -------
+    indicies : (num_cluster,) list of array of integer
+        The indicies of population that belong to each cluster.
+    """
+    logger.info("Clustering method is stripes.")
+    xp = cp.get_array_module(population)
+    # Sort the population along the dimension, then split into ranges of approx
+    # equal size
+    return np.array_split(
```

Comment:
Add a test that checks if this is stable. i.e. If we split into 1 cluster, the order of the population shouldn't change.